src/Pages/ItemDetail/Components/Board.js

```diff
@@ -1,48 +1,125 @@
 import React, { Component } from "react";
-import "./Board.scss";
+import { REVIEW_BOARD_API, INQUIRY_BOARD_API, REVIEW_MOCK, INQUIRE_MOCK } from "../../../config";
+import "./Style/Board.scss";
 
 const BOARD_NAME = {
   4: "Review",
   5: "Inquire",
 };
-const PAGES = ["<", 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, ">"];
+
+const LIMIT_PER_PAGE = 7;
+const PAGES_NUM = 5;
+const INITIAL_PAGES = Array.from({ length: PAGES_NUM }, (_, i) => i + 1);
+
 class Board extends Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      boardData: [],
-    };
-  }
+  state = {
+    boardData: [],
+    showForm: false,
+    currentPage: 1,
+    pages: INITIAL_PAGES,
+  };
+
+  postArticle = async () => {
+    /* 보드 종류별로 다른 API에 저장해야 한다. */
+    const response = await fetch(REVIEW_BOARD_API);
+    const result = await response.json();
+    await console.log(result);
+  };
+
+  showWriteForm = () => {
+    const { showForm } = this.state;
+    if (showForm) {
+      this.setState({ showForm: false });
+      this.postArticle();
+    } else {
+      this.setState({ showForm: true });
+    }
+  };
+
+  clickPage = e => {
+    this.getReviewData(e.target.id - 1);
+    this.setState({ currentPage: +e.target.id });
+  };
+
+  goPrevPage = () => {
+    const { currentPage, pages } = this.state;
+    console.log("go Prev Page");
+    if (currentPage === 1) return;
+    else if (currentPage % PAGES_NUM === 1) {
+      const newPages = pages.map(page => page - PAGES_NUM);
+      this.setState({ pages: newPages, currentPage: newPages[PAGES_NUM - 1] });
+    } else {
+      this.setState({ currentPage: currentPage - 1 });
+    }
+    this.getBoardData(currentPage - 1);
```

Comment:
return이 함수를 종료한다는 성질을 활용해서
```suggestion
    if (currentPage === 1) return;
    if (currentPage % PAGES_NUM === 1) {
      const newPages = pages.map(page => page - PAGES_NUM);
      this.setState({ pages: newPages, currentPage: newPages[PAGES_NUM - 1] });
      return;
    }
    this.setState({ currentPage: currentPage - 1 });
    this.getBoardData(currentPage - 1);
```

Reply:
감사합니다! 적용했습니다 👍